sdk/core/src/http_client/reqwest.rs

```diff
@@ -0,0 +1,82 @@
+use super::*;
+use std::{collections::HashMap, str::FromStr};
+
+/// Construct a new `HttpClient` with the `reqwest` backend.
+pub fn new_http_client() -> std::sync::Arc<dyn HttpClient> {
+    std::sync::Arc::new(::reqwest::Client::new())
+}
+
+#[async_trait]
+impl HttpClient for ::reqwest::Client {
+    async fn execute_request2(&self, request: &crate::Request) -> crate::Result<crate::Response> {
+        let url = request.url().clone();
+        let mut req = self.request(try_from_method(request.method())?, url);
+        for (name, value) in request.headers().iter() {
+            req = req.header(name.as_str(), value.as_str());
+        }
+
+        let body = request.body().clone();
+
+        let reqwest_request = match body {
+            Body::Bytes(bytes) => req
+                .body(bytes)
+                .build()
+                .context(ErrorKind::Other, "failed to build request")?,
+            Body::SeekableStream(mut seekable_stream) => {
```

Comment:
I'm interested to see if [http_types::Body]([http_types](https://docs.rs/http-types/latest/http_types/index.html)::[Body](https://docs.rs/http-types/latest/http_types/struct.Body.html#)) would provide benefits.